### PR TITLE
Add system warm-reboot check in WarmRestart class

### PR DESCRIPTION
**Pull request by zhenggen-xu**
Add system warm-reboot check in WarmRestart class
UT test is also updated

Add the WarmRestart Class to python binding

### CODE REVIEW

**Comment by jipanyang** [comment]
The change looks ok to me.  What is the use case?
One possible issue is that,  for processes in a docker which are experiencing warm restart,  the flag doesn't completely identify whether it is a system warm reboot or a docker warm restart.

Ex.  for swss docker restart,  if the processes see system warm restart flag, it will do warm data reload. But seeing the flag enabled doesn't mean system level reboot is in progress, it may just be a docker warm restart.

**Comment by zhenggen-xu** [comment]
This PR is for the system level warm reboot related processes. E,g, neighbor table restore during the system warm reboot.  We will check the system level reboot flag and make distinction between pure docker warm restart and system level warm reboot.  
C++/Python program will use this new lib function for checking instead of writing own functions.  It will be part of another PR update shortly.

**Comment by zhenggen-xu** [comment]
Also added the WarmRestart class to python binding

File: common/warm_restart.cpp [hunk 1]
```diff
@@ -173,6 +175,13 @@ bool WarmStart::isWarmStart(void)
     return warmStart.m_enabled;
 }
 
+bool WarmStart::isSystemWarmReboot(void)
```
Comment (jipanyang) [inline common/warm_restart.cpp]: Would it be better renaming it to isSystemWarmRebootEnabled() or something similar?  isSystemWarmReboot() seems  indicating the the system in going through warm reboot which may not be the case.

Comment (jipanyang) [inline common/warm_restart.cpp]: Would it be better renaming it to isSystemWarmRebootEnabled() or something similar? isSystemWarmReboot() seems indicating the the system in going through warm reboot which may not be the case.

**Comment by zhenggen-xu** [comment]
Done the renaming, also added a flag to increase the restore count or not.

File: common/warm_restart.cpp [hunk 2]
```diff
@@ -122,15 +124,14 @@ bool WarmStart::checkWarmStart(const std::string &app_name,
         warmStart.m_stateWarmRestartTable->hset(app_name, "restore_count", "0");
         return false;
     }
-    else
+
+    if (incr_restore_cnt)
```
Comment (jipanyang) [inline common/warm_restart.cpp]: I think "else if" should be used here.

**Pull request merged**